contributors/devel/sig-node/separate-mount-namespace.md

```diff
@@ -0,0 +1,212 @@
+day-to-day operation, there is no need for these special-use mounts to be
+visible in the host OS's top-level 'mount' or 'findmnt' output.
+
+### Security
+
+While access to container-specific mounts in Kubernetes (like secrets) today
+are protected by filesystem permissions, the fact that the locations are
+visible to unprivileged users via the 'mount' command can provide information
+which could be used in conjunction with other vulnerabilities to decipher
+secrets or other protected container mount contents. Locking access to even
+listing the mountpoints behind the ability to enter the mount namespace (which
+in turn requires `CAP_SYS_CHROOT` and `CAP_SYS_ADMIN`) removes one level of
+potential access to container-specific mounts.
+
+### No one uses it (that we know of)
+
+After much investigation and conversation on both the sig-node and sig-storage
+mailing lists, as well as within Red Hat's OpenShift communities, we have yet
+to turn up a use case that relies on this container-to-host propagation.
+
+### No one _should_ use it
+
+In fact, it may be seen as an anti-pattern and run contrary to the
+container-native philosophy. Relying on a host OS to interact with a
+container-owned mountpoint seems suspect.
+
+### Testing implications
+
+With separate mount namespaces, it would be possible to run multiple instances
+of Kubernetes on a single host without the various mountpoints interfering with
+one another. There may be other inter-instance difficulties besides the mount
+namespace, but this does move in that direction without resorting to more
+complicated solutions like trying to run Kubernetes inside of containers.
+
+## Costs and Implications
+
+If a Kubernetes distribution intends to implement this mount-hiding mode of
+operation, there are a number of costs and implications that should be
+documented for system administrators.
+
+### Privileged pods
+
+Privileged pods that previously expected to have direct access to the host's
+root mount namespace (such as the one spawned by `oc debug`), will now be
+rooted in the new container-runtime mount namespace.  They will still have
+visibility into the host OS's mount points as granted by the normal volumeMount
+permissions, but any mounts made in these privileged pods will not by default
+propagate up to be visible by the Host OS. In order for shells or utilities in
+privileged pods like this to propagate mounts up to the host OS, they can enter
+the host OS namespace by running `nsenter -t 1 -m` (assuming they have the host
+OS's /proc filesystem mounted so they can reach the mount namespace of the real
+init process in /proc/1/ns/mnt)
+
+### Legacy Tools
+
+Any monitoring tool that relies on the ability to run in the host OS and have
+visibility of mountpoints created by Kubelet, the Container Runtime, or
+containers themselves, will need to enter the container mount namespace in
+order to see these mountpoints.  This could be done by:
+
+- Altering the legacy tool to be cloud-native
```

Comment:
What does this mean?

Reply:
"Put it in a container", but I agree this can be clearer.